src/bt_peer_connection.cpp

```diff
@@ -1999,12 +1999,6 @@ namespace {
 			// if we are super seeding, pretend to not have any piece
 			// and don't send a bitfield
 			m_sent_bitfield = true;
-
-			// bootstrap super-seeding by sending two have message
-			piece_index_t piece = t->get_piece_to_super_seed(get_bitfield());
-			if (piece >= piece_index_t(0)) superseed_piece(piece_index_t(-1), piece);
-			piece = t->get_piece_to_super_seed(get_bitfield());
-			if (piece >= piece_index_t(0)) superseed_piece(piece_index_t(-1), piece);
```

Comment:
how does the "bootstrapping" happen now? what triggers us to send the first `HAVE` message?

Reply:
I found that bootstrapping here could fail. `superseed_piece()` requires `in_handshake()` to be false, whereas it can still be true here when `write_bitfield()` is called from `on_receive_impl()`.

'Bootstrapping' now happens in `second_tick()` when we decide whether to increase the number of pieces (ie. from 0 to the minimum of 2) once we're no longer in_handshake().